decompiler/config/all-types.gc

```diff
@@ -15872,40 +15874,41 @@
   ((game-task uint8   :offset-assert 0)
    (status    uint64  :offset-assert 8)
    (flags     uint8   :offset-assert 16)
-   (condition basic   :offset-assert 20)
+   (condition (function symbol symbol)   :offset-assert 20) ; TODO these are likely the lambdas
    )
   :method-count-assert 16
   :size-assert         #x18
   :flag-assert         #x1000000018
   (:methods
-    (dummy-9 () none 9)
-    (dummy-10 () none 10)
-    (dummy-11 () none 11)
-    (dummy-12 () none 12)
-    (dummy-13 () none 13)
-    (dummy-14 () none 14)
-    (dummy-15 () none 15)
+    (get-game-task (_type_) uint 9)
+    (get-task-status (_type_) uint 10)
+    (TODO-11 (_type_ symbol) symbol 11) ;; TODO - symbols are guesses, as is the `condition` field!
+    (first-flag-bit? (_type_) symbol 12)
+    (third-flag-bit? (_type_) symbol 13)
+    (TODO-RENAME-14 (_type_) int 14)
+    (clear-all-but-first-flag-bit (_type_) int 15)
     )
   )
 
 (deftype task-control (basic)
   ((current-stage int16  :offset-assert 4)
-   (stage         basic  :offset-assert 8)
+   (current-stage32 int32 :offset 4) ;; I added this, in the top level, it tries to access the first field as an int32
```

Comment:
I don't think this is right. If you remove this, and add this type cast, it should work:

```
  "(top-level-login task-control)" :[
    [165, "v1", "symbol"]
  ],
```
(will require https://github.com/water111/jak-project/pull/668)

The array contains both `task-control`s and `symbol`s, and in this case it was accessing the `type` field of a `symbol`.